packages/flutter/lib/src/material/badge.dart

```diff
@@ -57,11 +57,13 @@ class Badge extends StatelessWidget {
     this.child,
   });
 
-  /// Convenience constructor for creating a badge with a numeric
-  /// label with 1-3 digits based on [count].
+  /// Convenience constructor for creating a badge with a numeric label based on [count].
   ///
-  /// Initializes [label] with a [Text] widget that contains [count].
-  /// If [count] is greater than 999, then the label is '999+'.
+  /// Initializes [label] with a [Text] widget that shows:
+  /// - the [count] value if it is less than or equal to [maxCount],
+  /// - otherwise, shows '[maxCount]+'.
+  ///
+  /// For example, if [count] is 1000 and [maxCount] is 99, the label will display '99+'.
   Badge.count({
     super.key,
     this.backgroundColor,
```

Comment:
@QuncCccccc
Would it be possible to add a semanticsLabel to the Badge widget to improve accessibility?

Currently, the badge may not be accessible to screen readers, especially when it includes dynamic content like counts or status indicators.